app/models/template.rb

```diff
@@ -90,13 +101,14 @@ class Template < ApplicationRecord
 
   validates :family_id, presence: { message: PRESENCE_MESSAGE }
 
-
   # =============
   # = Callbacks =
   # =============
 
-  before_validation :set_defaults
-
+  # TODO: leaving this in for now, as this is better placed as an after_update than
+  # overwriting the accessors.  We want to ensure this template is published
+  # before we remove the published_version
+  # That being said, there's a potential race_condition where we have multiple-published-versions
   after_update :reconcile_published, if: -> (template) { template.published? }
```

Comment:
Just wanted to make sure the logic of the above comment makes sense, and check if there's a better approach

Reply:
makes sense to leave this one as is